### ISSUE TITLE
akka.http.scaladsl.ClientServerSpec.The low-level HTTP infrastructure should complete a request/response over https when request has `Connection: close` set failing on 2.13

### ISSUE THREAD

**Issue by jrudolph** [akka/akka-http#2532]
https://jenkins.akka.io:8498/job/akka-http-nightly/487/AKKA_VERSION=default,Node=akka-http-nightly-node,SCALA_VERSION=2.13.0-RC1,jdk=AdoptOpenJDK%208/testReport/junit/akka.http.scaladsl/ClientServerSpec/The_low_level_HTTP_infrastructure_should_complete_a_request_response_over_https_when_request_has__Connection__close__set/

```
Error Message

timeout (3 seconds) waiting for 1 messages on WarningFilter(None,Right(Hostname verification failed),false)

Stacktrace

java.lang.AssertionError: timeout (3 seconds) waiting for 1 messages on WarningFilter(None,Right(Hostname verification failed),false)
	at akka.testkit.EventFilter.intercept(TestEventListener.scala:119)
	at akka.http.scaladsl.ClientServerSpec.$anonfun$new$62(ClientServerSpec.scala:611)
	at akka.stream.testkit.scaladsl.StreamTestKit$.assertAllStagesStopped(StreamTestKit.scala:32)
	at akka.stream.testkit.Utils$.assertAllStagesStopped(Utils.scala:24)
	at akka.http.scaladsl.ClientServerSpec.$anonfun$new$61(ClientServerSpec.scala:588)
```

**Comment by johanandren** [comment]
Also on 2.11 in a nightly this weekend:
https://jenkins.akka.io:8498/job/akka-http-nightly-release-10.1/AKKA_VERSION=default,Node=akka-http-nightly-node,SCALA_VERSION=2.11.12,jdk=AdoptOpenJDK%208/944/testReport/junit/akka.http.scaladsl/ClientServerSpec/The_low_level_HTTP_infrastructure_should_complete_a_request_response_when_request_has__Connection__close__set/

**Comment by johanandren** [comment]
And 2.12
https://jenkins.akka.io:8498/job/akka-http-nightly/AKKA_VERSION=master,Node=akka-http-nightly-node,SCALA_VERSION=2.12.11,jdk=AdoptOpenJDK%208/190/

**Comment by jrudolph** [comment]
https://jenkins.akka.io:8498/job/akka-http-nightly/245/AKKA_VERSION=master,Node=akka-http-nightly-node,SCALA_VERSION=2.12.12,jdk=AdoptOpenJDK%208/testReport/junit/akka.http.scaladsl/ClientServerSpec/The_low_level_HTTP_infrastructure_should_complete_a_request_response_over_https_when_request_has__Connection__close__set/

**Comment by jrudolph** [comment]
This seems to be an actual issue which I documented in #3452. In this case we might be able to tweak options in a way that the issue doesn't turn up.

**Comment by ennru** [comment]
Fixed with #3456
